### PR TITLE
fix(checks_loader): solve issue related with checks from compliance

### LINKED ISSUE #5595: The compliance flag dosent gives key error when using
### Steps to Reproduce

Run prowler 4.5.0 with `--complinace` flag

### Expected behavior

The scan should proceed

### Actual Result with Screenshots or Logs

image

### How did you install Prowler?

Cloning the repository from github.com (git clone)

### Environment Resource

EKS 

### OS used

Amazon Linux

### Prowler version

v4.5.0

### Pip version

N/A

### Context

_No response_

### PR DESCRIPTION
**Pull request by pedrooot**
### Context
Fix #5595 

### Description
This pull request includes several changes to the `prowler` library, focusing on improving the flexibility and robustness of the `load_checks_to_execute` function and its related tests. The most important changes include making function parameters optional, updating the handling of compliance frameworks, and removing unnecessary test patches.

### Functionality Improvements:

* [`prowler/lib/check/checks_loader.py`](diffhunk://#diff-32e2ec5bfdd5944707b7ea1197ae3c7312a3696e9a203ee35639479f9558eccfL10-R18): Made parameters of the `load_checks_to_execute` function optional to enhance flexibility.
* [`prowler/lib/check/checks_loader.py`](diffhunk://#diff-32e2ec5bfdd5944707b7ea1197ae3c7312a3696e9a203ee35639479f9558eccfR28): Added a conditional check for `bulk_checks_metadata` to prevent errors when it is not provided.
* [`prowler/lib/check/checks_loader.py`](diffhunk://#diff-32e2ec5bfdd5944707b7ea1197ae3c7312a3696e9a203ee35639479f9558eccfL91-R92): Corrected a parameter name in the call to `CheckMetadata.list` to ensure the correct metadata is used.

### Codebase Simplification:

* [`prowler/lib/check/models.py`](diffhunk://#diff-8f998b586c18e08a5fa76f7508cf19c71088a5b28b392989c46882ad5c853eeaL190-R194): Changed the initialization of several dictionaries to sets to simplify the storage and retrieval of checks. [[1]](diffhunk://#diff-8f998b586c18e08a5fa76f7508cf19c71088a5b28b392989c46882ad5c853eeaL190-R194) [[2]](diffhunk://#diff-8f998b586c18e08a5fa76f7508cf19c71088a5b28b392989c46882ad5c853eeaL226-R226)

### Test Improvements:

* [`tests/lib/check/check_loader_test.py`](diffhunk://#diff-843325f3001b1e7e7c0768499b8c1568764c8d903b831308d316dc13657a7f21L66-L69): Removed unnecessary patches in multiple test cases to streamline the tests and ensure they accurately reflect the updated function behavior. [[1]](diffhunk://#diff-843325f3001b1e7e7c0768499b8c1568764c8d903b831308d316dc13657a7f21L66-L69) [[2]](diffhunk://#diff-843325f3001b1e7e7c0768499b8c1568764c8d903b831308d316dc13657a7f21L142-L145) [[3]](diffhunk://#diff-843325f3001b1e7e7c0768499b8c1568764c8d903b831308d316dc13657a7f21L172-L178) [[4]](diffhunk://#diff-843325f3001b1e7e7c0768499b8c1568764c8d903b831308d316dc13657a7f21L204) [[5]](diffhunk://#diff-843325f3001b1e7e7c0768499b8c1568764c8d903b831308d316dc13657a7f21L235-L238) [[6]](diffhunk://#diff-843325f3001b1e7e7c0768499b8c1568764c8d903b831308d316dc13657a7f21L254-L268)

### Checklist

- Are there new checks included in this PR? Yes / No
    - If so, do we need to update permissions for the provider? Please review this carefully.
- [x] Review if the code is being covered by tests.
- [x] Review if code is being documented following this specification https://github.com/google/styleguide/blob/gh-pages/pyguide.md#38-comments-and-docstrings
- [x] Review if backport is needed.

### License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: prowler/lib/check/checks_loader.py [hunk 1]
```diff
@@ -7,15 +7,15 @@
...
-    bulk_compliance_frameworks: dict,
-    checks_file: str,
-    check_list: list,
-    service_list: list,
-    severities: list,
-    compliance_frameworks: list,
-    categories: set,
-    provider: str,
+    bulk_checks_metadata: dict = None,
+    bulk_compliance_frameworks: dict = None,
```
Comment (jfagoagas) [inline prowler/lib/check/checks_loader.py:13]: I think these should be always required, what do you think?

File: tests/lib/check/check_loader_test.py [hunk 2]
```diff
@@ -232,51 +217,56 @@ def test_load_checks_to_execute_with_service_list(
...
+                    )
+                ],
+            ),
         }
-        bulk_compliance_frameworks = None
+        bulk_checks_metatada = None
         checks_file = None
         check_list = []
         service_list = []
         severities = []
```
Comment (jfagoagas) [inline tests/lib/check/check_loader_test.py]: If they are optional you can remove them right?

Comment (jfagoagas) [reply tests/lib/check/check_loader_test.py]: Review the rest of tests.

**Review by jfagoagas — CHANGES_REQUESTED** [review]

File: tests/lib/check/check_loader_test.py (hunk 2 above)
Comment (pedrooot) [reply tests/lib/check/check_loader_test.py]: Done!!

File: prowler/lib/check/checks_loader.py (hunk 1 above)
Comment (pedrooot) [reply prowler/lib/check/checks_loader.py:13]: I've made both optionals, the only one that it's really needed is the `bulk_checks_metadata`, that will fill all the dicts from `check_aliases`, `check_categories`, `check_severities` I've tried to get the bulk checks metadata only if a category or severity it's passed but the main problem resides on the alias from the check. We always update the checks with them aliases so check metadata it's needed.

Comment (jfagoagas) [reply prowler/lib/check/checks_loader.py:13]: Thanks for the explanation!

**Review by jfagoagas — APPROVED** [review]
Great fix, thanks for this one!

**Pull request merged**